fq: hide password/tokens in {List,Describe}Connection and logs

yumkam opened the pull request:
this is a sensitive field, same as password

### Changelog entry <!-- a user-readable short description of the changes that goes to CHANGELOG.md and Release Notes -->

...

### Changelog category <!-- remove all except one -->

* Not for changelog (changelog entry is not required)

### Description for reviewers <!-- (optional) description for those who read this PR -->

...

yumkam committed:
fq: don't list auth { token { token } } in {List,Describe}Connection

this is sensitive field, same as password

yumkam committed:
fixup! don't list auth { token { token } } in {List,Describe}Connection

yumkam committed:
fixup! fq: don't list auth { token { token } } in {List,Describe}Connection

yumkam committed:
Hide secrets in request logs

uzhastik commented on ydb/core/fq/libs/common/util.h:
```diff
@@ -75,6 +75,8 @@ EYdbComputeAuth GetYdbComputeAuthMethod(const FederatedQuery::ConnectionSetting&
 
 FederatedQuery::IamAuth GetAuth(const FederatedQuery::Connection& connection);
 
+FederatedQuery::IamAuth* GetMutableAuth(FederatedQuery::ConnectionSetting* setting);
```
давай на вход передадим ссылку, а не указатель

uzhastik commented on ydb/core/fq/libs/control_plane_storage/ydb_control_plane_storage_connections.cpp:
```diff
@@ -26,6 +28,9 @@ void PrepareSensitiveFields(::FederatedQuery::Connection& connection, bool extra
         pg.set_password("");
     }
+    if (auto auth = GetMutableAuth(&setting); auth && auth->has_token()) {
+        auth->mutable_token()->set_token("");
```
а нет готового метода для зачистки? auth->clear_token() ?

yumkam committed:
GetMutableAuth: pass parameter by reference

yumkam committed:
replace set_password("") with clear_password()

yumkam committed:
if -> switch

makes harder to forgot to add server kind here (mysql and greenplum was
overlooked)

yumkam committed:
fixup! ... and secure few missing potentially leaky logs

yumkam committed:
Partial Revert "Hide secrets in request logs"

This reverts commit 6594b6b18314c60e3f53755d252671aa23e44a4b.
Only secure messages that can contain secrets

yumkam committed:
... and secure few missing potentially leaky logs
